addons/web/static/src/core/l10n/numbers.js

```diff
@@ -86,11 +88,139 @@ export function parseFloat(value) {
         decimalPoint: decimalPointRegex,
     });
     if (isNaN(parsed)) {
-        throw new Error(sprintf(_lt("'%s' is not a correct float").toString(), value));
+        throw new ParsingError(sprintf(_lt("'%s' is not a correct float").toString(), value));
+    }
+    return parsed;
+}
+
+/**
+ * @param {string} value the string to be parsed
+ * @returns float
+ * @throws {Error} Error if the value can't be parsed properly
+ */
+export function parseFloatTime(value) {
+    let floatTime;
+    try {
+        let factor = 1;
+        if (value[0] === "-") {
+            value = value.slice(1);
+            factor = -1;
+        }
+        const values = value.split(":");
+        if (values.length === 1) {
+            floatTime = factor * parseFloat(value);
+        } else if (values.length === 2) {
+            const hours = parseInteger(values[0]);
+            const minutes = parseInteger(values[1]);
+            floatTime = factor * (hours + minutes / 60);
+        } else {
+            throw new ParsingError();
+        }
+    } catch (e) {
+        if (e instanceof ParsingError) {
+            throw new ParsingError(
+                sprintf(_lt("'%s' is not a correct float time").toString(), value)
+            );
+        } else {
+            throw e;
+        }
+    }
+    return floatTime;
+}
+
+/**
+ * Parse a more readable formated string representing an integer to an integer
```

Comment:
this docstring does not make sense to me. I am more confused now than before reading it